code/modules/antagonists/gang/gang_items.dm

```diff
@@ -0,0 +1,188 @@
+
+/obj/item/storage/box/shuriken_box
+	name = "shuriken Box"
+
+/obj/item/storage/box/shuriken_box/PopulateContents()
+	for (var/i in 1 to 4)
+		new /obj/item/throwing_star(src)
+
+/datum/gang_item/weapon/pistol
+	name = "10mm Pistol"
+	desc = "A small, easily concealable 10mm handgun. Has a threaded barrel for suppressors."
+	cost = 750
+	item_path = /obj/item/gun/ballistic/automatic/pistol
+
+/datum/gang_item/weapon/pistol_ammo
+	name = "10mm Ammo"
+	desc = "A low velocity magazine for 10mm pistols, ineffective against armour."
+	cost = 50
+	item_path = /obj/item/ammo_box/magazine/m10mm/lv
+
+/datum/gang_item/weapon/uzi
+	name = "Uzi SMG"
+	desc = "A lightweight submachine gun, for when you really want someone dead. Uses 9mm rounds."
+	cost = 1000
+	item_path = /obj/item/gun/ballistic/automatic/mini_uzi
+
+/datum/gang_item/weapon/uzi_ammo
+	name = "Uzi Ammo"
+	desc = "A 9mm magazine intended for use with Uzi SMGs."
+	cost = 70
+	item_path = /obj/item/ammo_box/magazine/uzim9mm
+
+/datum/gang_item/weapon/laser
+	name = "Laser Gun"
+	desc = "An older model of the basic lasergun, no longer used by Nanotrasen's private security or military forces. Nevertheless, it is still quite deadly and easy to maintain, making it a favorite amongst pirates and other outlaws."
+	cost = 600
+	item_path = /obj/item/gun/energy/laser/retro
+
+///////////////////
+//EQUIPMENT
+///////////////////
+
+/datum/gang_item/support //Anything that doesn't fit in the other categories.
+	category = "Support Equipment"
+
+/datum/gang_item/support/healcigs
+	name = "Healing Cigs"
+	desc = "A pack of omnizine laced syndicate cigarettes."
+	cost = 20
+	item_path = /obj/item/storage/fancy/cigarettes/cigpack_syndicate
+
+/datum/gang_item/support/maintpass
+	name = "Maintenance Access Pass"
+	desc = "A small card, that when used on an ID, will grant basic maintenance access."
+	cost = 10
+	item_path = /obj/item/card/id/pass/maintenance
+
+/datum/gang_item/support/medkit
+	name = "Basic Medical Kit"
+	desc = "A small card, that when used on an ID, will grant basic maintenance access."
+	cost = 30
+	item_path = /obj/item/storage/firstaid/regular
+
+/datum/gang_item/support/smuggler
+	name = "Smuggler's Satchel"
+	desc = "A small satchel, able to fit into tight spaces such as backpacks or under the floor boards."
+	cost = 50
+	item_path = /obj/item/storage/backpack/satchel/flat
+
+/datum/gang_item/infrep //You use these to win
+	category = "Influence & Reputation"
+
+/datum/gang_item/infrep/drugs
+	name = "Illicit Drug Dispenser"
+	desc = "Dispenses various kinds of narcotics, including formaltenamine, which grants influence and reputation when consumed by non gangsters."
+	cost = 100
+	item_path = /obj/item/sbeacondrop/drugs
+
+/datum/gang_item/infrep/drugs/spawn_item(mob/living/carbon/user, datum/team/gang/gang, obj/item/device/gangtool/gangtool)
+	var/obj/item/sbeacondrop/drugs/O = new item_path(user.loc)
+	user.put_in_hands(O)
+	O.g = gang
```

Comment:
and then, doing this:
```DM
	var/obj/item/sbeacondrop/drugs/O = ..()
	// user.put_in_hands(O) // no need
	O.g = gang
```

This can be a way if you just need to assign gang value (at parent proc)
```DM
//snip
	if(item_path)
		var/obj/item/O = new item_path(user.loc)
		user.put_in_hands(O)
		if(O.vars["gang_team"]) // as long as you don't name it just var/g for those...
				O.vars["gang_team"] = gang
```
This works well for those conditional items, then you don't have to do assigning values in subtype procs.

Reply:
Using vars is highly not recommended due to the loss of type safety and should only be used when there is no other way. Using : should be preferred over using .vars["var_name"].

Reply:
It looks it needs GetComponent because it needs to remember which gang team it refers to, and I thought it's better than a component.